src/map/pc.c

```diff
@@ -12030,14 +12008,25 @@ void pc_update_job_and_level(struct map_session_data *sd)
 	}
 }
 
-void do_final_pc(void) {
+void do_final_pc(void)
+{
+	int i, k;
+
 	db_destroy(pc->itemcd_db);
 	pc->at_db->destroy(pc->at_db,pc->autotrade_final);
 
 	pcg->final();
 
 	pc->clear_skill_tree();
 
+
+	for (k = 0; k < 2; k++) {
```

Comment:
this probably better move to separate function. Something like pc_clear_exp_groups.